src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/NativeProjectSystemReferencesReader.cs

```diff
@@ -108,5 +108,11 @@ internal class NativeProjectSystemReferencesReader
         {
             throw new NotSupportedException();
         }
+
+        public async Task<IReadOnlyList<(string id, string[] metadata)>> GetItemsAsync(string itemTypeName, params string[] metadataNames)
+        {
```

Comment:
Same here, do we know for sure this API is implemented?

Reply:
When debugging suppressions for a native c++ project with packages.config, it defintaely went into this codepath, and I saw in the debugger the suppressions were there. I don't know of any packages for native projects with vulnerabilities, so it's not easy to manually test. I don't remember if we have any Apex or VS E2E tests for native C++ projects. We could potentially add one, but NuGet doesn't really have huge usage in the C++ ecosystem, and it's even less likely that a native library with a known vulnerability will have an advisory on a nuget package.

Reply:
> but NuGet doesn't really have huge usage in the C++ ecosystem

I think my concern was more on the side of ensuring we're not crashing the restore.

Reply:
`VsProjectAdapter.GetBuildItemInformation` returns an empty result if the project system doesn't implement `IVsBuildItemStorage`

https://github.com/NuGet/NuGet.Client/blob/46f80a3f1a83f3c5f33a20fa9fb72871e1e2d4eb/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs#L228-L229

so, no crash.